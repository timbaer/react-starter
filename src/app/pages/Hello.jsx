import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const Hello = ({ user, onLoadRandomUser }) => (
    <section>
        Hello
        {' '}
        <Link to="/world">World</Link>
        {' '}
        {user ? ` to ${user.name}` : ''}
        <p>
            <button type="button" onClick={onLoadRandomUser}>Load User</button>
        </p>
    </section>
);


Hello.propTypes = {
    user: PropTypes.shape({
        id: PropTypes.number.isRequired,
        name: PropTypes.string.isRequired
    }),
    onLoadRandomUser: PropTypes.func.isRequired
};

Hello.defaultProps = {
    user: null
};

export default Hello;
