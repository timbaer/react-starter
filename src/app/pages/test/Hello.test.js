import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import Hello from '../Hello';


test('should render as expected with no user loaded', () => {
    const onLoadFn = jest.fn();

    const component = renderer.create(
        <MemoryRouter>
            <Hello onLoadRandomUser={onLoadFn} />
        </MemoryRouter>
    );
    const tree = component.toJSON();

    expect(tree)
        .toMatchSnapshot();
});

test('should render as expected with loaded user', () => {
    const onLoadFn = jest.fn();

    const component = renderer.create(
        <MemoryRouter>
            <Hello onLoadRandomUser={onLoadFn}
                user={{
                    id: 1,
                    name: 'Hans Wurst'
                }}
            />
        </MemoryRouter>
    );
    const tree = component.toJSON();

    expect(tree)
        .toMatchSnapshot();
});
