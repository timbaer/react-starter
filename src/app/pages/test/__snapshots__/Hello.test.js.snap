// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render as expected with loaded user 1`] = `
<section>
  Hello
   
  <a
    href="/world"
    onClick={[Function]}
  >
    World
  </a>
   
   to Hans Wurst
  <p>
    <button
      onClick={[MockFunction]}
      type="button"
    >
      Load User
    </button>
  </p>
</section>
`;

exports[`should render as expected with no user loaded 1`] = `
<section>
  Hello
   
  <a
    href="/world"
    onClick={[Function]}
  >
    World
  </a>
   
  
  <p>
    <button
      onClick={[MockFunction]}
      type="button"
    >
      Load User
    </button>
  </p>
</section>
`;
